### PR TITLE
#213 Donation page

### PR DESCRIPTION
**Pull request by Toxicom**
Issue #213 
Issue https://app.forecast.it/project/P-59/workflow/T9173

# Summary of Changes

1. Create `<Icon />` component
2. Refactor `<ContactChipSections />` to `<TitledSection />`
   - That component can be used not only for contact chips
3. Use `<Icon />` component in `<TitledSection />` and `<SlidingNavigation />`
4. Refactor `/aukojimas/` subpages to a single page
5. Remove `/shared-pages/Aukojimas/`
   - This is not needed anymore since that route is the same single page without subpages
1. Rename `<Constraint />` modifier from `--contact-chip-sections` to `--sliding-nav` to be more universal
2. Refactor `<SlidingNavigation />` intersection observer:
   - Add clean-up function on unmount;
   - Handle cases where an observed element never left the screen.

## Unrelated changes
1. Improve `--header-height` variable by adding media queries to it
   - This acts as a fallback in case there's no JS

# Notes
N/a

## To do
- [x] Code clean up *(eg.: delete `shared-pages/Aukojimas`)*
- [x] Use links in sliding nav

# Screenshots

## Wide layout
image
image

## Narrow layout
image
image

# To test

- [ ] Go to `/kaip-galiu-padeti/aukojimas`
- [ ] See if everything works as expected

### CODE REVIEW

File: src/components/TitledSection/TitledSection.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        </h2>
+      </div>
+      <div className="TitledSection__content">{children}</div>
+    </div>
+  );
+};
+
+TitledSection.propTypes = {
+  id: PropTypes.string,
+  title: PropTypes.string,
```
Comment (grinkus-adapt) [inline src/components/TitledSection/TitledSection.js:23]: `isRequired`?

**Review by grinkus-adapt — CHANGES_REQUESTED** [review]

File: src/components/TitledSection/TitledSection.js (hunk 1 above)
Comment (Toxicom) [reply src/components/TitledSection/TitledSection.js]: 2711a0b0fb36907a925f3fe20c17d71bcffc3559

File: src/components/SlidingNavigation/SlidingNavigation.js [hunk 2]
```diff
@@ -62,12 +74,16 @@ const SlidingNavigation = ({ data, options }) => {
...
+    return () => {
+      observer.disconnect();
+    };
   }, []);
 
   const handleNavigationClick = (e, linkId) => {
     const element = document.getElementById(linkId);
-    const headerHeight = document.querySelector(`.Header`).offsetHeight || 66; // 66 fallback
-    const navbarHeight = navigationRef.current.offsetHeight || 63; // 63 fallback
+    const headerHeight = document.querySelector(`.Header`).offsetHeight || 88; // 88 fallback
```
[the remark this answers is not in the document]
Comment (Toxicom) [reply src/components/SlidingNavigation/SlidingNavigation.js]: 95118d9ff2a8952fe0017242d177068a6ec1708e

**Comment by Toxicom** [comment]
Looks like 95118d9 build failed, but it works locally. Next commit 2cb2fc1 had no `js` changes and it also got built correctly. Not sure why it failed that one time on cloudflare :thinking: 
image

**Pull request merged**